vms/evm/metrics/prometheus/prometheus_test.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package prometheus
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ava-labs/libevm/metrics"
+	"github.com/prometheus/common/expfmt"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/vms/evm/metrics/metricstest"
+)
+
+func TestGatherer_Gather(t *testing.T) {
+	metricstest.WithMetrics(t)
+
+	registry := metrics.NewRegistry()
+	register := func(t *testing.T, name string, collector any) {
+		t.Helper()
+		err := registry.Register(name, collector)
+		require.NoErrorf(t, err, "registering collector %q", name)
+	}
+
+	counter := metrics.NewCounter()
+	counter.Inc(12345)
+	register(t, "test/counter", counter)
+
+	counterFloat64 := metrics.NewCounterFloat64()
+	counterFloat64.Inc(1.1)
+	register(t, "test/counter_float64", counterFloat64)
+
+	gauge := metrics.NewGauge()
+	gauge.Update(23456)
+	register(t, "test/gauge", gauge)
+
+	gaugeFloat64 := metrics.NewGaugeFloat64()
+	gaugeFloat64.Update(34567.89)
+	register(t, "test/gauge_float64", gaugeFloat64)
+
+	gaugeInfo := metrics.NewGaugeInfo()
+	gaugeInfo.Update(metrics.GaugeInfoValue{"key": "value"})
+	register(t, "test/gauge_info", gaugeInfo) // skipped
+
+	sample := metrics.NewUniformSample(1028)
+	histogram := metrics.NewHistogram(sample)
+	register(t, "test/histogram", histogram)
+
+	meter := metrics.NewMeter()
+	t.Cleanup(meter.Stop)
+	meter.Mark(9999999)
+	register(t, "test/meter", meter)
+
+	timer := metrics.NewTimer()
+	t.Cleanup(timer.Stop)
+	timer.Update(20 * time.Millisecond)
+	timer.Update(21 * time.Millisecond)
+	timer.Update(22 * time.Millisecond)
+	timer.Update(120 * time.Millisecond)
+	timer.Update(23 * time.Millisecond)
+	timer.Update(24 * time.Millisecond)
+	register(t, "test/timer", timer)
+
+	resettingTimer := metrics.NewResettingTimer()
+	register(t, "test/resetting_timer", resettingTimer)
+	resettingTimer.Update(time.Second) // must be after register call
+
+	emptyResettingTimer := metrics.NewResettingTimer()
+	register(t, "test/empty_resetting_timer", emptyResettingTimer)
+
+	emptyResettingTimer.Update(time.Second) // no effect because of snapshot below
+	register(t, "test/empty_resetting_timer_snapshot", emptyResettingTimer.Snapshot())
```

Comment:
Would it make sense to move this out to a helper function `registerRealMetrics` like how we do with `registerNilMetrics`? I feel like it is a bit hard to actually see what this function is testing currently.